rules/eslint/suggestions.js

```diff
@@ -0,0 +1,213 @@
+    'no-octal': 'error',
+    'no-octal-escape': 'error',
+    'no-param-reassign': ['error', { props: false }],
+    'no-plusplus': ['error', { allowForLoopAfterthoughts: true }],
+    'no-proto': 'error',
+    // handled by "no-var" and "no-shadow" rules
+    'no-redeclare': 'off',
+    'no-regex-spaces': 'error',
+    // no restricted exports by default
+    'no-restricted-exports': 'off',
+    // no restricted globals by default
+    'no-restricted-globals': 'off',
+    // no restricted imports by default
+    'no-restricted-imports': 'off',
+    // no specific properties to restrict
+    'no-restricted-properties': 'off',
+    'no-restricted-syntax': [
+      'error',
+      {
+        selector: 'ForInStatement',
+        message: 'for..in loops are slower than Object.{keys,values,entries} and have their caveats.',
+      },
+      // From eslint-config-airbnb
+      // https://github.com/airbnb/javascript/blob/b4377fb03089dd7f08955242695860d47f9caab4/packages/eslint-config-airbnb-base/rules/style.js#L333
+      {
+        selector: 'LabeledStatement',
+        message: 'Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.',
+      },
+      {
+        selector: 'WithStatement',
+        message: '`with` is disallowed in strict mode because it makes code impossible to predict and optimize.',
+      },
+    ],
+    'no-return-assign': ['error', 'always'],
+    'no-return-await': 'error',
+    'no-script-url': 'error',
+    'no-sequences': 'error',
+    'no-shadow': 'error',
+    'no-shadow-restricted-names': 'error',
+    // ternaries are very useful
+    'no-ternary': 'off',
+    'no-throw-literal': 'error',
+    'no-undef-init': 'error',
+    // undefined is useful
+    'no-undefined': 'off',
+    // disabled by default; enable if necessary
+    'no-underscore-dangle': 'off',
+    'no-unneeded-ternary': 'error',
+    'no-unused-expressions': 'error',
+    // handled by "no-labels" rule
+    'no-unused-labels': 'off',
+    'no-useless-call': 'error',
+    'no-useless-catch': 'error',
+    'no-useless-computed-key': 'error',
+    // handled by "prefer-template" rule
+    'no-useless-concat': 'off',
+    'no-useless-constructor': 'error',
+    'no-useless-escape': 'error',
+    'no-useless-rename': 'error',
+    'no-useless-return': 'error',
+    'no-var': 'error',
+    'no-void': 'error',
+    // it is necessary to do warning comments sometimes
+    'no-warning-comments': 'off',
+    'no-with': 'error',
+    'object-shorthand': ['error', 'always'],
+    'one-var': ['error', 'never'],
+    // handled by Prettier
+    'one-var-declaration-per-line': 'off',
+    'operator-assignment': ['error', 'always'],
+    'prefer-arrow-callback': ['error', { allowNamedFunctions: false, allowUnboundThis: false }],
+    'prefer-const': ['error', { ignoreReadBeforeAssign: true }],
+    'prefer-destructuring': ['error', { array: false, object: true }, { enforceForRenamedProperties: true }],
```

Comment:
This rule seems to be too aggressive. 

- With `enforceForRenamedProperties: true`, ESLint reports 1326 problems in the web-components repo. 
- With `enforceForRenamedProperties: false`, ESLint reports 150 problems. 

Maybe we should disable `enforceForRenamedProperties` as the following change doesn't actually bring much value:

```diff
- const element = document.body.firstElementChild;
+ const { firstElementChild: element } = document.body;
```

Reply:
Well, I usually use the second option, because your concern valid only when the element is single. When you want to have two elements from the same object, it becomes easier to use the destructuring pattern:
```ts
const { firstElementChild: element,  className: cls } = document.body;
```
That's why this rule is introduced in the first place. Since it has the `--fix` option, I don't think it would require too much work to handle all the cases.

Reply:
According to the discussion, we decided that it's preferable to keep `enforceForRenamedProperties` as `false`.